src/arch/x86/archx86.rs

```diff
@@ -726,16 +677,8 @@ pub fn disasmx86<T: ArchDetail + Debug>(
         address: _instructions.address,
     };
     let mut instr = Instructionx86::new(code_info);
-    let ret: bool = match _mode {
```

Comment:
The only reason I lifted this translation between `Mode` and `Modex86` elsewhere is because on first read I didn't even notice the parameter was of different type (`::Mode` vs `::arch::x86::archx86::Modex86`), so these match arms looked oddly duplicative. It *also* removes a point where `x86` would potentially have to know about non-x86 things, but I'm not sure how important that is.